src/System.Management.Automation/utils/Telemetry.cs

```diff
@@ -563,19 +563,56 @@ private static bool GetEnvironmentVariableAsBool(string name, bool defaultValue)
                 return defaultValue;
             }
 
-            switch (str.ToLowerInvariant())
+            var boolStr = str.AsSpan();
```

Comment:
It looks to me `AsSpan` is not needed. You can just use `str` for the subsequent character comparisons.

Reply:
> It looks to me `AsSpan` is not needed. You can just use `str` for the subsequent character comparisons.

JIT optimization is a little bit [more favorable with span](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHjykDYBAgAbBgEFcAZQAO2AHYAKfgAYGuDFDRCR4gCYwAZtgCuojADVso0zACUHLjxcA3bFF1jpWhgF4NLQA6KTlFJXsAbhpnF24ASyMGJWFvYIAZGAUAcwwAC38A8ntYuM5qOMqGROTU0R8oAG1VAF1ChgBycg6SiqqXcv7+4gB2Bi07aL6hhgBfGOmhmpS9Bua2vwCO1R7S/sGZuNGGE1FcGCnDub2eeeobhKSVtKggzJz89sYAMm+HuOe9S063aHQAnh0GIRCF4gU1WqCAJo9Bi/f4uQFrcgbLYwSHQ2FYnGdACiKLRi36mOBpGJHVw+JhdTWtNB0h6vX26J4xwmF3+d3+x0MJnMVhsk1KdxcpVK9CYAlhkhkWniORU5HUmm0SpFZgs1lsDiclJ47k8zN8AW1U3+y0tr3euQKmwYxX+Bxm9tWwIRrq6u1NlU9VyYYz5l0OgqDcW9LxB/p2nMOIaux1O50jM2lVWj/TjcLeWWdX1RfxjGIdCa2EKhTJ98LpyMcFKu1Ka2NBeLrhOBnf9ZJb5bbVdZ/oZPdHdPZ9mTwe53F5UElMbzVWFxn14qNWeu1FmQA=) due to eliding bounds checks.  Unless that was also added to string and just hasn't hit sharplab yet.

(Not making a statement of worth, just making sure the difference is known)

Reply:
Indeed. Bounds check is removed for the `span` case once you start the check with `boolStr.Length == ...`, but not for the `string` case. This is good to know, thanks @SeeminglyScience.

Reply:
Closed?